lib/rules/web/console/base/login.xyaml

```diff
@@ -15,15 +15,15 @@ login_sequence:
     if_element:
       selector: &pre_login_buttons
         # we just choose the first button per OPENSHIFTQ-12062
-        css: ".login-redhat:first-of-type, .idp:first-of-type"
+        xpath: //a[text()='<idp>']
```

Comment:
I am concerned about this, that it will not work in case we don't know the idp name. I wonder if we can keep pushing first button in case IDP name not specified. WDYT?

Reply:
I agree to always click the first IDP when no IDP is set, how can I update the code?

Reply:
Appears that matching based on condition is tricky [1].

Another option is for the method `Environment#openshift_idp` method to execute API calls to figure out the IDP in case such is not provided in configuration. Is this possible?

[1] https://stackoverflow.com/questions/971067/is-there-an-if-then-else-statement-in-xpath

Reply:
Need to play with `curl` and API to see if this is visible.

Reply:
I realize that maybe there's issue if we execute API calls to get IDP when it's not provided, this operation will not be allowed for online, right?

Reply:
@yapei  definitely you can not. Online ENV is owned by Ops team, I believe in future the 'admin' idp will not be exposed out.

Reply:
@akostadinov  Does it mean we can't run API calls to IDP with above limitation?

Reply:
@akostadinov  Can you please take a look at this?

Reply:
It is not about discussing. It is more about trying it out.

Reply:
maybe it will be easier if we have `xpath: //a[text()='<idp>' or contains(@class, 'idp') or contains(@class, 'login-redhat')][0]`, wdyt?

The only issue I see is that in case `idp` is specified but not matched, then it would pick up first IDP, thus potential issues with IDP name might be missed.